library/general/src/modules/PackagesProposal.rb

```diff
@@ -44,20 +46,23 @@ def main
       #          ]
       #      ]
       @resolvables_to_install = {}
+      # the same as above but the resolvables are considered optional
+      @opt_resolvables_to_install = {}
 
       # List of currently supported types of resolvables
       @supported_resolvables = [:package, :pattern]
     end
 
-    # Resets all resolvables to install. Use carefully.
+    # Resets all resolvables to install (both required and optional). Use carefully.
     def ResetAll
-      if @resolvables_to_install != {}
-        Builtins.y2warning("Reseting all PackagesProposal items")
+      if !@resolvables_to_install.empty? || !@opt_resolvables_to_install.empty?
+        log.warn("Resetting all PackagesProposal items")
```

Comment:
why it is warning? you call method to reset it and it warn if it reset something?